### PR TITLE
Fixing broken TokenClient SDK

### PR DESCRIPTION
**Pull request by hariria**
## Motivation

The TokenClient for the Typescript SDK was broken. Summary of the issues are listed below for future reference:
1. The Tables API was missing on the AptosClient and missing functions like `getTableItem` which is needed for getting collections and tokens. 
2. Tests were failing for the TokenClient because of incorrect payload arguments and incorrect url fetching. For example `create_limited_token_script` was being used where it should have been `create_unlimited_token_script`. Furthermore, `/tables/${tableHandle}/items` should have been `/tables/${tableHandle}/item` for fetching table items
3. Functions were outdated (ie. `getTokenId` was replaced with `getTokenBalance` and `getTokenData`)
4. TokenClient is now at 100% coverage. 

We should discuss ways to prevent this in the future as there are multiple potential points of failure and takes time to debug. A few suggestions:

1. Stronger type-checking on requests and responses from the sdk
2. Stronger error checking, more useful error messaging from the sdk
3. Ensuring CI/CD runs testing properly
4. Maybe we could create rust / typescript interface bindings for functions to ensure uniform behavior across all language implementations
5. Pass params as dictionaries (this is more explicit) and will remove confusion around ordering of params
```javascript
// instead of 
async getTokenBalance(creator: MaybeHexString, collectionName: string, tokenName: string) { ... }

// do this
interface GetTokenBalanceProps {
  creator: MaybeHexString;
  collectionName: string;
  tokenName: string;
}

async getTokenBalance({ 
  creator,
  collectionName, 
  tokenName 
}: GetTokenBalanceProps) { ... }
```

Will Run NPM Publish after PR is landed and merged

## Test Plan

Before

```bash
 PASS  src/util.test.ts
 PASS  src/hex_string.test.ts
 PASS  src/aptos_account.test.ts
 PASS  src/aptos_client.test.ts
 FAIL  src/token_client.test.ts (5.569 s)
 PASS  src/faucet_client.test.ts (7.333 s)
------------------|---------|----------|---------|---------|-------------------
File              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------|---------|----------|---------|---------|-------------------
All files         |   89.44 |    86.66 |   94.33 |   89.65 |                   
 aptos_account.ts |     100 |      100 |     100 |     100 |                   
 aptos_client.ts  |   97.53 |       90 |     100 |    97.5 | 232,247           
 faucet_client.ts |     100 |      100 |     100 |     100 |                   
 hex_string.ts    |     100 |      100 |     100 |     100 |                   
 token_client.ts  |   55.26 |        0 |   66.66 |   54.28 | 80-139,151-163    
 util.ts          |     100 |      100 |     100 |     100 |                   
------------------|---------|----------|---------|---------|-------------------
```

Now
```bash
 PASS  src/hex_string.test.ts
 PASS  src/aptos_account.test.ts
 PASS  src/util.test.ts
 PASS  src/aptos_client.test.ts
 PASS  src/faucet_client.test.ts (7.123 s)
 PASS  src/token_client.test.ts (15.559 s)
------------------|---------|----------|---------|---------|-------------------
File              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------|---------|----------|---------|---------|-------------------
All files         |   99.46 |    96.42 |     100 |   99.45 |                   
 aptos_account.ts |     100 |      100 |     100 |     100 |                   
 aptos_client.ts  |   98.82 |       95 |     100 |    98.8 | 251               
 faucet_client.ts |     100 |      100 |     100 |     100 |                   
 hex_string.ts    |     100 |      100 |     100 |     100 |                   
 token_client.ts  |     100 |      100 |     100 |     100 |                   
 util.ts          |     100 |      100 |     100 |     100 |                   
------------------|---------|----------|---------|---------|-------------------

Test Suites: 6 passed, 6 total
Tests:       26 passed, 26 total
Snapshots:   0 total
Time:        16.09 s
Ran all test suites.
✨  Done in 16.95s.
```

### CODE REVIEW

**Review by CapCap — APPROVED** [review]

File: ecosystem/typescript/sdk/src/api/Tables.ts [hunk 1]
```diff
@@ -25,11 +25,11 @@ export class Tables<SecurityDataType = unknown> {
    * @tags state, table
    * @name GetTableItem
    * @summary Get table item by handle and key.
-   * @request POST:/tables/{table_handle}/items
+   * @request POST:/tables/{table_handle}/item
```
[the remark this answers is not in the document]
Comment (msmouse) [reply ecosystem/typescript/sdk/src/api/Tables.ts:28]: And https://github.com/aptos-labs/aptos-core/pull/818 tries to fix it.. you can fix it in this as well, if you want.

**Review by msmouse — APPROVED** [review]

File: ecosystem/typescript/sdk/src/api/Tables.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hariria) [reply ecosystem/typescript/sdk/src/api/Tables.ts:28]: No worries!

**Pull request merged**